queries/tlaplus/highlights.scm

```diff
@@ -160,3 +212,21 @@
 (proof_step_ref (level) @number)
 (proof_step_ref (name) @constant)
 (proof_step_ref ">" @punctuation.bracket)
+
+; Comments and tags
+(block_comment "(*" @comment)
+(block_comment "*)" @comment)
+(block_comment_text) @comment
+(comment) @comment
+(single_line) @comment
+(_ label: (identifier) @tag)
+(pcal_goto statement: (identifier) @tag)
+
+; Reference highlighting with the same color as declarations.
+; `constant`, `operator`, and others are custom captures defined in locals.scm
+((identifier_ref) @constant.builtin (#is? @constant.builtin constant))
+((identifier_ref) @operator (#is? @operator operator))
+((identifier_ref) @parameter (#is? @parameter parameter))
+((identifier_ref) @variable.builtin (#is? @variable.builtin builtin_variable))
+((identifier_ref) @variable.builtin (#is? @variable.builtin variable))
```

Comment:
Please don't use `#is?` for now. It will work but our current implementation is to slow. You can out-comment them with a TODO. If TLA files tend to be rather small the performance issue might not be relevant. In that case, you can leave them here. It would be good to test edits on a few files to make sure that there are no performance problems.

Reply:
I am aware of the `#is?` usage performance concerns. In practice, TLA files tend to be no longer than 1k lines. I've tested this while editing all the examples from [this](https://github.com/tlaplus/Examples/tree/master/specifications) repo and haven't noticed any lags